### ISSUE TITLE
Support iOS style swipable table cells

### ISSUE THREAD

**Issue by xster** [flutter/flutter#116303]
Given that table / list item swipe is a common part of the iOS design language

![Transition-Reveal](https://user-images.githubusercontent.com/156888/204916281-7277e22a-7223-4ea6-8875-2c2b6e910ea8.gif)

and also part of the standard UIKit APIs through https://developer.apple.com/documentation/uikit/uiswipeactionsconfiguration and https://developer.apple.com/documentation/uikit/uicontextualaction, we should support them in Cupertino. 

We should also be consistent about affordance of these Cupertino functionalities. Either all Cupertino widgets are in individual Flutter Favorite packages or all Cupertino widgets (part of the UIKit API) are in the Cupertino package itself.

**Comment by Piinks** [comment]
@darrenaustin thinks we already have this implemented somewhere in a ListItem/ListTile-ish widget.. I haven't been able to find it yet.

I think it would be really nice to have this available as a general widget that can wrap any other widget and enable this swipe to reveal behavior. That way we won't have to re-implement it in a bunch of different widgets where this behavior may be desired. :)

**Comment by darrenaustin** [comment]
The widget I was thinking about is [Dismissible](https://api.flutter.dev/flutter/widgets/Dismissible-class.html). However, I don't think it supports icon buttons.

There is also a community package that looks provides this kind of UI: [flutter_slidable](https://pub.dev/packages/flutter_slidable).

**Comment by xster** [comment]
Right, it's already there in flutter_slidable. The meta-point is more so that we should be consistent. Either everything spec'ed in HIG or trivially achievable via UIKit APIs should be bundled by default in Cupertino or none should be. Having an arbitrary subset be in Cupertino while some being 3p packages reduces search affordance. It could work if we came up with some "rule" that makes it simple for users to predict whether something will be in Cupertino or not. But I haven't been able to come up with anything that slices neatly to include things like pull to refresh but not this.
